### PR TITLE
Add `allow_search` property to ItemList and Tree to control incremental search

### LINKED ISSUE #4682: Option to disable incremental search on ItemList node
### Describe the project you are working on

An inventory system using keyboard navigation.
image

### Describe the problem or limitation you are having in your project

Pressing any letter key causes the ItemList node to search the list for a result who's name matches the keypress. This behaviour is outside the common understanding of the node as well as the documentation, and as such should be disabled by default.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

The option to enable/disable key searching should appear in the inspector in the ItemList settings. 

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

image

### If this enhancement will not be used often, can it be worked around with a few lines of script?

There is currently no workaround

### Is there a reason why this should be core and not an add-on in the asset library?

Incremental search being enabled by default goes against the documentation and adds unnecessary functionality while removing versatility. It should be removed or made optional.

### PR DESCRIPTION
**Pull request by needleful**
Implementing https://github.com/godotengine/godot-proposals/issues/4682. 

This makes Tree and ItemList consistent with PopupMenu, which has the allow_search property to control incremental search, which is sometimes not desired.  Default is true to preserve existing behavior.

This was needed for my own game, since I wanted the player to open a tree menu while playing to spawn entities, but input was read by the incremental search and interfered with navigation.

I've done basic testing for both nodes with and without search to ensure it works as expected.

_Production edit: Closes https://github.com/godotengine/godot-proposals/issues/4682_.

### CODE REVIEW

File: doc/classes/ItemList.xml [hunk 1]
```diff
@@ -341,6 +341,9 @@
 		<member name="allow_rmb_select" type="bool" setter="set_allow_rmb_select" getter="get_allow_rmb_select" default="false">
 			If [code]true[/code], right mouse button click can select items.
 		</member>
+		<member name = "allow_search" type="bool" setter="set_allow_search" getter="get_allow_search" default="true">
+			If [code]true[/code], allows navigating the [ItemList] with letter keys through incremental search.
```
Comment (YuriSizov) [inline doc/classes/ItemList.xml:345]:
```suggestion
		<member name="allow_search" type="bool" setter="set_allow_search" getter="get_allow_search" default="true">
			If [code]true[/code], allows navigating the [ItemList] with letter keys through incremental search.
```

File: doc/classes/Tree.xml [hunk 2]
```diff
@@ -335,6 +335,9 @@
 		<member name="allow_rmb_select" type="bool" setter="set_allow_rmb_select" getter="get_allow_rmb_select" default="false">
 			If [code]true[/code], a right mouse button click can select items.
 		</member>
+		<member name = "allow_search" type="bool" setter="set_allow_search" getter="get_allow_search" default="true">
```
Comment (YuriSizov) [inline doc/classes/Tree.xml:338]:
```suggestion
		<member name="allow_search" type="bool" setter="set_allow_search" getter="get_allow_search" default="true">
```

**Comment by YuriSizov** [comment]
Try not to edit the structure of class ref files manually. Use `doctool` instead. See [this article](https://docs.godotengine.org/en/stable/contributing/documentation/updating_the_class_reference.html#updating-class-reference-when-working-on-the-engine).

**Review by KoBeWi — APPROVED** [review]

**Review by akien-mga — APPROVED** [review]
The feature makes sense to me.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**